### PR TITLE
feat: refactor

### PR DESCRIPTION
**Pull request by przemyslaw-grobecki**

### CODE REVIEW

File: typescript-codegen.ts [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+const primitivesGenerationPath = `${workingDirectory}/PokeClient.ts/generated/primitives`;
+const resourcesGenerationPath = `${workingDirectory}/PokeClient.ts/generated/resources`;
+
+const codegenHelper : TypescriptCodegenHelper = new TypescriptCodegenHelper();
+codegenHelper.LoadPrimitives(primitivesSchemasPath).then((val) => {
+    codegenHelper.GeneratePrimitives(primitivesGenerationPath);
+});
+codegenHelper.LoadResources(resourcesSchemasPath).then((val) => {
+    codegenHelper.GenerateResources(resourcesGenerationPath);
+});
```
Comment (przemyslaw-grobecki) [inline typescript-codegen.ts:241]: This goes concurrently.... whyyy???

File: typescript-codegen.ts [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+                    fileContent,
+                    jsonContent,
+                    typescriptInterfaceContent
+                )
+            );
+        });
+        this.resourcesPath = resourcesPath;
+    };
+
+    public GenerateResources = async (generationPath: string) => {
```
Comment (krzkro4122) [inline typescript-codegen.ts:166]: You're running async callbacks, that are just thrownn at the js event loop and executed on worker threads asynchronously and independedly, so concurrently.

Do you intend to sync them in some way?

File: typescript-codegen.ts (hunk 1 above)
Comment (krzkro4122) [reply typescript-codegen.ts:241]: Wdym? You wanna sync the Loads/GenerateResources in some way?

File: typescript-codegen.ts (hunk 2 above)
Comment (przemyslaw-grobecki) [reply typescript-codegen.ts:166]: There is no need to mark this func as async. This is just the product of some blind shots to overcome the issue in the bottom of the file.

**Pull request closed without merging**